### PR TITLE
MM-10136: Not allowing to edit others posts in the interface

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Not allowing to edit others posts in the interface

#### Ticket Link
[MM-10136](https://mattermost.atlassian.net/browse/MM-10136)

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: src/utils/post_utils.js [hunk 1]
```diff
@@ -72,7 +72,9 @@ export function canEditPost(state, config, license, teamId, channelId, userId, p
         if (hasNewPermissions(state)) {
             canEdit = canEdit && haveIChannelPermission(state, {team: teamId, channel: channelId, permission: Permissions.EDIT_POST});
             if (!isOwner) {
-                canEdit = canEdit && haveIChannelPermission(state, {team: teamId, channel: channelId, permission: Permissions.EDIT_OTHERS_POSTS});
+                // I only can edit my posts, at least until phase-2
```
Comment (lieut-data) [inline src/utils/post_utils.js]: I could sworn I had posted this comment already, but it vanished...

Is the only thing stopping this from working a client-side check? Do we need server changes to complement this change?

Comment (jespino) [reply src/utils/post_utils.js]: Same here, Server change not needed because it is intentionally disabled only in the interface.

**Comment by koxen** [comment]
are you aware it needs a fix on server side too ?

**Comment by lieut-data** [comment]
@koxen, @jespino and I discussed this above (inline). It turns out that the ability for an admin to edit posts was always in place, but the UI was incorrectly exposed (as it's not likely to be a common use-case). Going forward, my understanding is that advanced permissions is precisely about constraining such abilities as individual use cases demand.

**Pull request merged**